passes/opt/rmports.cc

```diff
@@ -74,7 +74,7 @@ struct RmportsPassPass : public Pass {
 			}
 
 			auto ports_to_remove = removed_ports[cell->type];
-			for(auto p : ports_to_remove)
+			for(const auto &p : ports_to_remove)
```

Comment:
That's a funny one! `auto ports_to_remove` obviously should be `const auto ports_to_remove&`, but isn't. But instead of fixing that we still copy the whole container of ints, and then patch the code so it creates a reference to each of the individual ints. Why?

Reply:
Oh, the reason why the `auto ports_to_remove` is not constified is because this automated change ( https://clang.llvm.org/extra/clang-tidy/checks/performance-for-range-copy.html ) is only concerned about range copies, not where the inputs are coming from. Agreed, that this would probably be a good idea :)

The `const auto &p` vs. `const auto p` will make a difference here if p is a string.